readme clean-up (#1872)
DataDog/datadog-operator: README.md

See the [Getting Started][5] dedicated documentation to learn how to deploy the Datadog operator and your first Agent, and [Configuration][12] to see examples, a list of all configuration keys, and default values.

### Migrating from `v1alpha1` to `v2alpha1`

> Current Operator version does not support migration from DatadogAgent CRD version `v1alpha1` to `v2alpha1` or from Operator `v0.8.x` to `v1.x.x`. 

Reviewer: ```suggestion
Current Operator version does not support migration from the Datadog Agent CRD version `v1alpha1` to `v2alpha1` or from Operator `v0.8.x` to `v1.x.x`. 
```

Reviewer 2: ```Datadog Operator v1.8.0+ does not support migrating from `DatadogAgent` CRD `v1alpha1` to `v2alpha1` or from Operator `v0.8.x` to `v1.x.x`. ```

> Please use `v1.7.0` to migrate and then upgrade to a latest version. 

Reviewer 3: ```suggestion
Please use the conversion webhook from `v1.7.0` to migrate and then upgrade to a latest version. 
```

> Please use `v1.7.0` to migrate and then upgrade to a latest version. 

Reviewer: ```suggestion
Use `v1.7.0` to migrate and then upgrade to a latest version. 
```

> Use the conversion webhook in v1.7.0 to migrate, and then upgrade to a recent version.

Reviewer 3: ```suggestion
Use the conversion webhook in `v1.7.0` to migrate, and then upgrade to a recent version.
```
